docs/usage/BOLFI.rst

```diff
@@ -0,0 +1,309 @@
+
+This tutorial is generated from a `Jupyter <http://jupyter.org/>`__
+notebook that can be found
+`here <https://github.com/elfi-dev/notebooks>`__.
+
+BOLFI
+-----
+
+In practice inference problems often have a complicated and
+computationally heavy simulator, and one simply cannot run it for
+millions of times. The Bayesian Optimization for Likelihood-Free
+Inference `BOLFI <http://jmlr.csail.mit.edu/papers/v17/15-017.html>`__
+framework is likely to prove useful in such situation: a statistical
+model (e.g. `Gaussian
+process <https://en.wikipedia.org/wiki/Gaussian_process>`__, GP) is
+created for the discrepancy, and its minimum is inferred with `Bayesian
+optimization <https://en.wikipedia.org/wiki/Bayesian_optimization>`__.
+This approach typically reduces the number of required simulator calls
+by several orders of magnitude.
+
+This tutorial demonstrates how to use BOLFI to do LFI in ELFI.
+
+.. code:: ipython3
+
+    import numpy as np
+    import scipy.stats
+    import matplotlib
+    import matplotlib.pyplot as plt
+    
+    %matplotlib inline
+    %precision 2
+    
+    import logging
+    logging.basicConfig(level=logging.INFO)
+    
+    # Set an arbitrary global seed to keep the randomly generated quantities the same
+    seed = 20170622
+    np.random.seed(seed)
+    
+    import elfi
+
+Although BOLFI is best used with complicated simulators, we will use the
+familiar MA2 model introduced in the basic tutorial, and load it from
+ready-made examples:
+
+.. code:: ipython3
+
+    from elfi.examples import ma2
+    model = ma2.get_model(seed_obs=seed)
+    elfi.draw(model)
+
+
+
+
+.. image:: http://research.cs.aalto.fi/pml/software/elfi/docs/0.5/usage/BOLFI_files/BOLFI_5_0.svg
+
+
+
+Fitting the surrogate model
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now we can immediately proceed with the inference. However, when dealing
+with a Gaussian process, it may be beneficial to take a logarithm of the
+discrepancies in order to reduce the effect that high discrepancies have
+on the GP. In ELFI such transformed node can be created easily:
```

Comment:
...on the GP. Sometimes you may want to add a small constant to avoid very negative or even -Inf distances occurring especially if it is likely that there can be exact matches between simulated and observed data.